src/Persistence/Models/Control.cs

```diff
@@ -50,15 +45,10 @@ public Control(string name)
     /// <summary>
     /// key/value pairs of Control properties. Mapped to/from Control rules.
     /// </summary>
-    [YamlMember(Order = 2)]
     public ControlPropertiesCollection Properties { get; init; } = new();
 
     /// <summary>
     /// list of child controls nested under this control.
-    /// </summary>
-    [YamlMember(Order = 3)]
-    public IList<Control> Controls { get; init; } = new List<Control>();
-
-    [YamlIgnore]
-    public ControlEditorState? EditorState { get; set; }
+    /// </summary>    
+    public Control[] Controls { get; init; } = Array.Empty<Control>();
```

Comment:
Perhaps out of scope of this PR:

Apps, Screens, and Canvas Components have child objects (Canvas Components and Screens both contain controls, Apps contain Screens and Canvas Components), but to my knowledge no other controls can have child controls.  Perhaps this property should be moved either directly to those types, or to somewhere else in the inheritance hierarchy.

Reply:
this is a good point. We're basically saying that only these three types of objects can have a tree-like structure (`Apps`, `Screens` and `Components` ). Everything else is flat.
Let's discuss with the architects first, I want to make sure we're covering all the scenarios. My point is that removing the ability to create a tree might backfire in the future and I want to avoid that.